BeeKit/UI/BSButton.swift

```diff
@@ -13,23 +13,52 @@ public class BSButton : UIButton {
     
     required init?(coder aDecoder: NSCoder) {
         super.init(coder: aDecoder)
+        
+        registerForTraitChanges(
+            [UITraitUserInterfaceLevel.self]) {
+                (self: Self, previousTraitCollection: UITraitCollection) in
+                self.resetStyle()
+            }
+        
         self.setUp()
     }
     
     override init(frame: CGRect) {
         super.init(frame: frame)
+        
+        registerForTraitChanges(
+            [UITraitUserInterfaceLevel.self]) {
+                (self: Self, previousTraitCollection: UITraitCollection) in
+                self.resetStyle()
+            }
```

Comment:
As this is duplicated in both init methods, perhaps it should instead live in setUp?